### PR TITLE
Make rcsetup.py flak8 compliant

### PR DESCRIPTION
**Pull request by hershen**
## PR Summary

Inspired by #12525, this PR makes rcsetup.py flake8 compliant.

### CODE REVIEW

File: lib/matplotlib/rcsetup.py [hunk 1]
```diff
@@ -74,7 +74,8 @@ def f(s):
                 if allow_stringlist:
                     # Sometimes, a list of colors might be a single string
                     # of single-letter colornames. So give that a shot.
-                    return [scalar_validator(v.strip()) for v in s if v.strip()]
+                    return [scalar_validator(v.strip()) for v in s if
+                            v.strip()]
```
Comment (timhoffm) [inline lib/matplotlib/rcsetup.py]: The preferred way to split list comprehensions is before the keyword `for` (and/or `if` if additionally required:

~~~
return [scalar_validator(v.strip())
        for v in s if v.strip()]
~~~

This keeps the logical blocks together.

File: lib/matplotlib/rcsetup.py [hunk 2]
```diff
@@ -837,20 +839,20 @@ def validate_cycler(s):
...
-            # might come from the internet (future plans), this
-            # could be downright dangerous.
-            # I locked it down by only having the 'cycler()' function
-            # available.
-            # UPDATE: Partly plugging a security hole.
-            # I really should have read this:
-            # http://nedbatchelder.com/blog/201206/eval_really_is_dangerous.html
-            # We should replace this eval with a combo of PyParsing and
-            # ast.literal_eval()
+           # TODO: We might want to rethink this...
```
Comment (timhoffm) [inline lib/matplotlib/rcsetup.py]: Why are you dedenting here? This was correctly indented 4 spaces.

File: lib/matplotlib/rcsetup.py [hunk 3]
```diff
@@ -1012,7 +1015,7 @@ def _validate_linestyle(ls):
     'lines.solid_joinstyle': ['round', validate_joinstyle],
     'lines.dash_capstyle':   ['butt', validate_capstyle],
     'lines.solid_capstyle':  ['projecting', validate_capstyle],
-    'lines.dashed_pattern':  [[3.7, 1.6], validate_nseq_float(allow_none=True)],
+    'lines.dashed_pattern': [[3.7, 1.6], validate_nseq_float(allow_none=True)],
```
Comment (timhoffm) [inline lib/matplotlib/rcsetup.py]: While generally, we try to be flake8 conform, I would not insist on line length limits for the definition of the default parameters. These definitions should be kept as structured as possible. I thus propose to leave E501 in and limit the changes to `defaultParams` to cases where we do not have to bent over or excessively stretch comments over serveral lines to satisfy the line limit.

**Comment by hershen** [comment]
Thanks @timhoffm. 
I removed the problematic changes.

**Comment by timhoffm** [comment]
Thanks @hershen and congratulations to your first contribution to matplotlib!

**Comment by hershen** [comment]
Thanks!

**Pull request merged**